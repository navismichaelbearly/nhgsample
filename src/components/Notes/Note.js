import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { formatting } from 'nhh-styles';

import { NoteBody, NoteDescription, NoteItem, NoteLabel } from './components';

export const NoteDictionaryPropType = {
  heading: PropTypes.string.isRequired,
  labels: PropTypes.shape({
    date: PropTypes.string.isRequired,
    filename: PropTypes.string.isRequired,
    fileUploadStatus: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
};

export const NotePropType = {
  created: PropTypes.shape({
    on: PropTypes.string.isRequired,
    by: PropTypes.string,
  }).isRequired,
  id: PropTypes.string.isRequired,
  subject: PropTypes.string,
  text: PropTypes.string,
};

const splitFileString = (str, key = 0) => {
  const item = str ? str.split('|')[key] : null;
  return item ? item.trim() : null;
};

export const Note = ({ note: { created, id, subject, text }, dictionary: { labels } }) => (
  <NoteItem key={id} data-bdd={`note-item-${id}`}>
    <NoteBody>
      <NoteLabel>{labels.date}</NoteLabel>{' '}
      <span data-bdd={`note-${id}-date`}>
        {formatting.formatDate(created.on, formatting.dateTimeFormat)}
      </span>
      <br />
      {created.by && (
        <Fragment>
          <NoteLabel>{labels.name}</NoteLabel>{' '}
          <span data-bdd={`note-${id}-name`}>{created.by}</span>
          <br />
        </Fragment>
      )}
      {splitFileString(subject, 1) && (
        <Fragment>
          <NoteLabel>{labels.filename}</NoteLabel>{' '}
          <span data-bdd={`note-${id}-filename`}>{splitFileString(subject, 1)}</span>
          <br />
        </Fragment>
      )}
      {splitFileString(subject, 2) && (
        <Fragment>
          <NoteLabel>{labels.fileUploadStatus}</NoteLabel>{' '}
          <span data-bdd={`note-${id}-fileUploadStatus`}>{splitFileString(subject, 2)}</span>
          <br />
        </Fragment>
      )}
      {text && <NoteDescription data-bdd={`note-${id}-text`}>{text}</NoteDescription>}
    </NoteBody>
  </NoteItem>
);

Note.propTypes = {
  dictionary: PropTypes.shape(NoteDictionaryPropType).isRequired,
  note: PropTypes.shape(NotePropType).isRequired,
};
