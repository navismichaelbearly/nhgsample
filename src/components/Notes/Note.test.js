import React from 'react';
import { shallow } from 'enzyme';
import * as nhhStyles from 'nhh-styles';
import { Note } from './Note';

nhhStyles.formatting.formatDate = jest.fn(() => '10/08/2018 - 8:52pm');

describe('<Note />', () => {
  let props;

  beforeEach(() => {
    props = {
      note: {
        id: 'e2054c02-08dd-e811-80d2-005056825b41',
        caseId: '00000000-0000-0000-0000-000000000000',
        created: { by: null, on: '2018-10-31T12:25:13+00:00' },
        modified: { by: null, on: '2018-10-31T12:25:13+00:00' },
      },
      dictionary: {
        heading: 'Notes ({count})',
        labels: {
          date: 'Date',
          filename: 'Filename',
          fileUploadStatus: 'FileUploadStatus',
          name: 'Name',
        },
      },
    };
  });

  it('renders correctly', () => {
    expect(shallow(<Note {...props} />)).toMatchSnapshot();
  });

  it('renders correctly with a filename', () => {
    const testProps = { ...props, note: { ...props.note, subject: 'blah | filename.jpg' } };
    expect(shallow(<Note {...testProps} />)).toMatchSnapshot();
  });

  it('renders correctly with a file upload status', () => {
    const testProps = {
      ...props,
      note: { ...props.note, subject: 'blah | filename.jpg | A status' },
    };
    expect(shallow(<Note {...testProps} />)).toMatchSnapshot();
  });

  it('renders correctly with text', () => {
    const testProps = {
      ...props,
      note: { ...props.note, text: 'some text' },
    };
    expect(shallow(<Note {...testProps} />)).toMatchSnapshot();
  });
});
