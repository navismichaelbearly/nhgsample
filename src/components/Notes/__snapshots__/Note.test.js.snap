// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Note /> renders correctly 1`] = `
<styled.div
  data-bdd="note-item-e2054c02-08dd-e811-80d2-005056825b41"
  key="e2054c02-08dd-e811-80d2-005056825b41"
>
  <styled.div>
    <styled.span>
      Date
    </styled.span>
     
    <span
      data-bdd="note-e2054c02-08dd-e811-80d2-005056825b41-date"
    >
      10/08/2018 - 8:52pm
    </span>
    <br />
  </styled.div>
</styled.div>
`;

exports[`<Note /> renders correctly with a file upload status 1`] = `
<styled.div
  data-bdd="note-item-e2054c02-08dd-e811-80d2-005056825b41"
  key="e2054c02-08dd-e811-80d2-005056825b41"
>
  <styled.div>
    <styled.span>
      Date
    </styled.span>
     
    <span
      data-bdd="note-e2054c02-08dd-e811-80d2-005056825b41-date"
    >
      10/08/2018 - 8:52pm
    </span>
    <br />
    <styled.span>
      Filename
    </styled.span>
     
    <span
      data-bdd="note-e2054c02-08dd-e811-80d2-005056825b41-filename"
    >
      filename.jpg
    </span>
    <br />
    <styled.span>
      FileUploadStatus
    </styled.span>
     
    <span
      data-bdd="note-e2054c02-08dd-e811-80d2-005056825b41-fileUploadStatus"
    >
      A status
    </span>
    <br />
  </styled.div>
</styled.div>
`;

exports[`<Note /> renders correctly with a filename 1`] = `
<styled.div
  data-bdd="note-item-e2054c02-08dd-e811-80d2-005056825b41"
  key="e2054c02-08dd-e811-80d2-005056825b41"
>
  <styled.div>
    <styled.span>
      Date
    </styled.span>
     
    <span
      data-bdd="note-e2054c02-08dd-e811-80d2-005056825b41-date"
    >
      10/08/2018 - 8:52pm
    </span>
    <br />
    <styled.span>
      Filename
    </styled.span>
     
    <span
      data-bdd="note-e2054c02-08dd-e811-80d2-005056825b41-filename"
    >
      filename.jpg
    </span>
    <br />
  </styled.div>
</styled.div>
`;

exports[`<Note /> renders correctly with text 1`] = `
<styled.div
  data-bdd="note-item-e2054c02-08dd-e811-80d2-005056825b41"
  key="e2054c02-08dd-e811-80d2-005056825b41"
>
  <styled.div>
    <styled.span>
      Date
    </styled.span>
     
    <span
      data-bdd="note-e2054c02-08dd-e811-80d2-005056825b41-date"
    >
      10/08/2018 - 8:52pm
    </span>
    <br />
    <styled.p
      data-bdd="note-e2054c02-08dd-e811-80d2-005056825b41-text"
    >
      some text
    </styled.p>
  </styled.div>
</styled.div>
`;
